pkg/kubelet/kubelet.go

```diff
@@ -1418,6 +1418,24 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 
 		// start syncing lease
 		go kl.nodeLeaseController.Run(wait.NeverStop)
+
+		// Wait for node cache be ready:
+		// if new labels are labeled in a node, and pods use nodeSelector,
+		// kubelet may reject the pods at kubelet restarting because node
+		// cache is not populated and kubelet still uses stale node info
+		// which has no new labels.
+		_ = wait.PollInfinite(10*time.Millisecond, func() (bool, error) {
```

Comment:
Infinite waits make me a little nervous :) Thoughts on a max amount of time we give the cache to populate before concluding it won't ever populate and either trying to move on or raising an error?

Reply:
@mattjmcnaughton if the node cache is not populated, if still moving on, the pod may fail at predicating; for raising error, I added log to remind the current status. This issue happens when kubeclient failed to contact apiserver.

This is the error from reflector of node ListWatch in a goroutine:
```
Nov 05 12:44:34 kubelet[14390]: E1105 12:44:34.893917 14390 reflector.go:134] k8s.io/kubernetes/pkg/kubelet/kubelet.go:452: Failed to list *v1.Node: Stream error http2.StreamError{StreamID:0x9, Code:0x2, Cause:error(nil)} when reading response body, may be caused by closed connection. Please retry.
```
This is the error when predicating pods:
```
Nov 05 12:44:34 kubelet[14390]: I1105 12:44:34.919779 14390 predicate.go:133] Predicate failed on Pod: xxx(a1bfbee0-5d58-11e9-adbb-ac1f6b0b668c), for reason: Predicate MatchNodeSelector failed
```
There is race condition between them.